blockchain/blockdao/indexbuilder.go

```diff
@@ -100,25 +100,36 @@ func (ib *IndexBuilder) handler() {
 		select {
 		case <-ib.cancelChan:
 			return
-		case blk := <-ib.pendingBlks:
-			timer := ib.timerFactory.NewTimer("indexBlock")
-			if err := ib.indexer.PutBlock(blk); err != nil {
+		default:
```

Comment:
define a func fetchBlockFromBuffer() (*block.Block, error), return next block (tipHeight+1) if it exists, otherwise nil

blk, err := fetchBlockFromBuffer()
if blk == nil || err != nil {
     continue
}